cdap-api/src/main/java/co/cask/cdap/api/templates/AdapterConfigurer.java

```diff
@@ -43,6 +44,13 @@
   public void setInstances(int instances);
 
   /**
+   * Set the resources the program should use.
+   *
+   * @param resources
+   */
+  public void setResources(Resources resources);
```

Comment:
In the case of ETL Adapters, will this be derived from the source, sink specification? For example, in Batch ETL Adapter, the source and sink set the resources required for Mapper through their context (there is a bug which needs to be fixed - to figure out if resources are already set and use the max). For realtime case, the source, sink specification will return resources which could then be used to implement the AdapterConfigurer#setResources method for ETL Realtime template. And so for Batch case we can return some constant?